src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs

```diff
@@ -360,6 +374,9 @@ public Task PublishUpdateAsync(IResource resource, string resourceId, Func<Custo
 
             var newState = stateFactory(previousState);
 
+            // Increment the snapshot version, this is a per resource version.
+            newState = newState with { Version = notificationState.LastVersion++ };
```

Comment:
nit: Make LastVersion a private field, and have a method called `GetNextVersion()` on notification state that increments and returns a value?

Reply:
This whole type is private 😄